utils/network.py

```diff
@@ -0,0 +1,107 @@
+import networkx as nx
+import pandas as pd
+
+from utils.linkage import deduplicate_perfect_matches
+
+
+def deduplicate_datasets(
+    ind_df: pd.DataFrame, org_df: pd.DataFrame, transactions_df: pd.DataFrame
+) -> tuple:
+    """Deduplicates the uuids in the inds and orgs dfs and updates the uuids in
+    transactions dataset to match those in the new inds and orgs dfs
+
+    Args:
+        ind_df: A pandas df with individual information
+        org_df: A pandas df with organization information
+        transactions df: A pandas df with info on transactions between entities
+
+    Returns:
+        A tuple of the ind_df, org_df, and transactions_df
+    """
+    # apply dedup to both inds and orgs
+    inds_df = deduplicate_perfect_matches(ind_df)
+    orgs_df = deduplicate_perfect_matches(org_df)
+
+    # update the deduplicated uuids in transaction donor and recipient columns
+    # to the uuids they are mapped to
+    deduped = pd.read_csv("../output/deduplicated_UUIDs.csv")
+    transactions_df[["donor_id", "recipient_id"]] = transactions_df[
+        ["donor_id", "recipient_id"]
+    ].replace(deduped)
+
+    return inds_df, orgs_df, transactions_df
+
+
+def name_identifier(uuid: str, orgs_df, inds_df) -> str:
+    """Returns the name of the entity given the entity's uuid
+
+    Args:
+        uuid: the uuid of the entity
+        orgs_df and inds_df: the dataframes from which the entities uuid
+        is queried
+
+    Return:
+        The entity's name
+    """
+    # first, check orgs df:
+    name_in_org = orgs_df.loc[orgs_df["id"] == uuid]
+    if len(name_in_org) > 0:
+        return name_in_org.iloc[0]["name"]
+    # theoretically it must be in inds if not in orgs, but for the sample data
+    # this might not be the case
+    name_in_ind = inds_df.loc[inds_df["id"] == uuid]
+    if len(name_in_ind) > 0:
+        return name_in_ind.iloc[0]["full_name"]
+    else:
+        return None
+
+
+def network_prep_pipeline(
+    ind_df: pd.DataFrame, org_df: pd.DataFrame, transactions_df: pd.DataFrame
+) -> tuple:
+    """Pipeline for preparing the orgs, inds, and transactions dataframes for
+    network linkage
+
+    Args:
+        ind_df, org_df, transactions_df: pandas dataframes with information
+        regarding campaign contributions between donors and recipients
+
+    Returns:
+        a tuple containing the 3 dataframes ready for network building
+    """
+
+    ind_df, org_df, transactions_df = deduplicate_datasets(
+        ind_df, org_df, transactions_df
+    )
+
+    # add recipient_name to the transactions dataset
```

Comment:
I think you can throw this in the function below or rename the function to be more specific to this purpose